File: backend/src/main/java/tom/projects/monkeylog/model/exercise/ExerciseType.java
package tom.projects.monkeylog.model.exercise;

import tom.projects.monkeylog.model.MetricFormat;

public enum ExerciseType {
    REPS("Reps", MetricFormat.NUMBER),
    TIME("Time", MetricFormat.TIME),
    DISTANCE("Distance", MetricFormat.DISTANCE),
    WEIGHT("Weight", MetricFormat.WEIGHT);

    private final String name;
    private final MetricFormat metricFormat;

    ExerciseType(String name, MetricFormat metricFormat) {
        this.name = name;
        this.metricFormat = metricFormat;
    }

    public String getName() {
        return name;
    }

    public MetricFormat getMetricFormat() {
        return metricFormat;
    }

}

File: backendkt/src/main/kotlin/tom/com/monkeylog/model/exercise/ExerciseType.kt
package tom.com.monkeylog.model.exercise

import tom.com.monkeylog.model.MetricFormat

enum class ExerciseType(val niceName: String, val metricFormat: MetricFormat) {
    REPS("Reps", MetricFormat.NUMBER),
    TIME("Time", MetricFormat.TIME),
    DISTANCE("Distance", MetricFormat.DISTANCE),
    WEIGHT("Weight", MetricFormat.WEIGHT);
}
